querydsl-core/src/main/java/com/querydsl/core/types/dsl/BooleanExpression.java

```diff
@@ -177,13 +177,27 @@ public BooleanExpression nullif(Boolean other) {
         return nullif(ConstantImpl.create(other));
     }
 
+    /**
+     * Create a {@code coalesce(this, expr)} expression
+     *
+     * @param expr additional argument
+     * @return coalesce
+     */
+    @Override
+    public BooleanExpression coalesce(Expression<Boolean> expr) {
+        Coalesce<Boolean> coalesce = new Coalesce<Boolean>(getType(), mixin);
+        coalesce.add(expr);
+        return coalesce.asBoolean();
+    }
+
     /**
      * Create a {@code coalesce(this, exprs...)} expression
      *
      * @param exprs additional arguments
      * @return coalesce
      */
     @Override
+    @SuppressWarnings("unchecked")
```

Comment:
Does this eliminate "SafeVarargs" warnings?

Reply:
In our codebase yes, not on the user side. To completely get rid of all warnings on user side we'd have to remove `T`

Reply:
I don't think removing <T> will solve it. The problem is that the method cannot be final right?

Anyways, the other two overloads will probably be the most frequently used ones.

Reply:
`Expression<?>...` would remove the warning on that method, but allows wrong parameters to be passed in.

Reply:
I'm not really seeing why `Expression<?>...` would prevent heap pollution because the array on the heap can still be exposed as return variable, but I am willing to sacrifice a bit of input type safety in exhange for not having to bother explaining our users that our methods are in fact SafeVarargs but cant be annotated as such ;-)

Reply:
I'm not saying it makes sense, it just works like this. Reverted the type adding commit on this branch.

Reply:
Seems its been to long since I've read Effective Java to understand SafeVarargs properly 😉